### PR TITLE
Implements the PlayerLoomPatternSelect event

### PR DESCRIPTION
**Pull request by heypr**

### CODE REVIEW

File: paper/src/main/java/com/denizenscript/denizen/paper/PaperModule.java [hunk 1]
```diff
@@ -63,6 +63,7 @@ public static void init() {
             ScriptEvent.registerScriptEvent(PlayerRaiseLowerItemScriptEventPaperImpl.class);
         }
         ScriptEvent.registerScriptEvent(PlayerSelectsStonecutterRecipeScriptEvent.class);
+        ScriptEvent.registerScriptEvent(PlayerLoomPatternSelectScriptEvent.class);
```
Comment (BreadcrumbIsTaken) [inline paper/src/main/java/com/denizenscript/denizen/paper/PaperModule.java]: Not in alphabetical order. I know it's not directly stated that it needs to be, but its kinda like an unspoken standard :P

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    //
+    // @Switch type:<pattern> to only process the event if the specified pattern is selected.
+    //
+    // @Context
+    // <context.loom> returns an InventoryTag of the loom.
+    // <context.pattern> returns an ElementTag of the selected pattern. Valid pattern types can be found at:
+    // <@link url https://jd.papermc.io/paper/1.19/org/bukkit/block/banner/PatternType.html>
+    //
+    // @Determine
+    // "TYPE:<ElementTag>" to set the pattern type of the loom.
```
Comment (BreadcrumbIsTaken) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java]: PATTERN

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            String lower = CoreUtilities.toLowerCase(determinationObj.toString());
+            if (lower.startsWith("pattern:")) {
+                ElementTag value = new ElementTag(lower.substring("pattern:".length()));
+                event.setPatternType(PatternType.valueOf(String.valueOf(value).toUpperCase()));
+                return true;
+            }
+        }
+        return super.applyDetermination(path, determinationObj);
+    }
+    @EventHandler
```
Comment (BreadcrumbIsTaken) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java:100]: There should be a new line separating these.

**Comment by heypr** [comment]
Done :)

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(event.getPlayer());
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "loom" -> new InventoryTag(event.getLoomInventory().getType());
```
Comment (tal5) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java]: You're constructing it from the type, which means it won't be a reference to the actual inventory - use `mirrorBukkitInventory`

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        };
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
+        if (determinationObj instanceof ElementTag) {
+            String lower = CoreUtilities.toLowerCase(determinationObj.toString());
+            if (lower.startsWith("pattern:")) {
+                ElementTag value = new ElementTag(lower.substring("pattern:".length()));
+                event.setPatternType(PatternType.valueOf(String.valueOf(value).toUpperCase()));
```
Comment (tal5) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java]: `ElementTag#asEnum`

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    // @Player Always.
+    //
+    // @Triggers when a player selects a loom pattern.
+    //
+    // @Switch type:<pattern> to only process the event if the specified pattern is selected.
+    //
+    // @Context
+    // <context.loom> returns an InventoryTag of the loom.
+    // <context.pattern> returns an ElementTag of the selected pattern. Valid pattern types can be found at:
+    // <@link url https://jd.papermc.io/paper/1.19/org/bukkit/block/banner/PatternType.html>
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLoomPatternSelectScriptEvent.java]: shouldn't have a newline here

**Pull request merged**